content/src/service/access/AccessCheckerImpl.ts

```diff
@@ -1,90 +1,399 @@
+         * at that time
+         */
+
+        const owner = estate.owners[0].address.toLowerCase()
+
+        const authorizations = await this.getAuthorizations(
+            owner,
+            ethAddress,
+            timestamp)
+
+        const firstOperatorAuthorization = authorizations.find(
+            authorization => authorization.type === 'Operator')
+        const firstApprovalForAllAuthorization = authorizations.find(
+            authorization => authorization.type === 'ApprovalForAll')
+
+        if (firstOperatorAuthorization?.isApproved || firstApprovalForAllAuthorization?.isApproved) {
+            return true
+        }
+
+        return false
+    }
+
+
+    private async getParcel(x: number, y: number, timestamp: Timestamp): Promise<Parcel> {
+        /**
+         * You can use `owner`, `operator` and `updateOperator` to check the current value for that parcel.
+         * Keep in mind that each association (owners, operators, etc) is capped to a thousand (1000) results.
+         * For more information, you can use the query explorer at https://thegraph.com/explorer/subgraph/nicosantangelo/watchtower
+         */
+
+        const query = `
+            query GetParcel($x: Int!, $y: Int!, $timestamp: Int!) {
+                parcels(where:{ x: $x, y: $y }) {
+                    estates(
+                            where: { createdAt_lte: $timestamp },
+                            orderBy: createdAt,
+                            orderDirection: desc,
+                            first: 1
+                        ) {
+                        estateId
+                    }
+                    owners(
+                            where: { createdAt_lte: $timestamp },
+                            orderBy: createdAt,
+                            orderDirection: desc,
+                            first: 1
+                        ) {
+                        address
+                    }
+                    operators(
+                            where: { createdAt_lte: $timestamp },
+                            orderBy: createdAt,
+                            orderDirection: desc,
+                            first: 1
+                        ) {
+                        address
+                    }
+                    updateOperators(
+                            where: { createdAt_lte: $timestamp },
+                            orderBy: createdAt,
+                            orderDirection: desc,
+                            first: 1
+                        ) {
+                        address
+                    }
+                }
+            }`
+
+        const variables = {
+            x, y,
+            timestamp: Math.floor(timestamp / 1000) // UNIX
+        }
+
+        try {
+            const response = await this.queryGraph<{ parcels: Parcel[] }>(
+                query,
+                variables
+            )
+            return response.parcels[0]
```

Comment:
Are we sure this will always return at least one element? If not, then this might return `undefined`

Reply:
fixed!